ui/app/models/secret-engine.js

```diff
@@ -49,7 +49,7 @@ export default Model.extend(Validations, {
   }),
   // KV 2 additional config default options
   maxVersions: attr('number', {
-    defaultValue: 10,
```

Comment:
api docs need to be updated. I double checked this by running the api commands and then reading the config. the default here is 0, though on a secret it's 10.